src/utils/createContainerShape.js

```diff
@@ -0,0 +1,15 @@
+import createAdaptorShape from './createAdaptorShape';
+
+export default function createContainerShape(PropTypes) {
+  const adaptorShape = createAdaptorShape(PropTypes);
+  return PropTypes.shape({
+    adaptor: adaptorShape.isRequired,
+    container: PropTypes.shape({
```

Comment:
containers could and probably should have a bigger API. I can imagine that 'dumb components' should become decorators that can perform some kind of projection on container state.